kaye_tech/backend/tabletop/character.py

```diff
@@ -0,0 +1,185 @@
+from dataclasses import dataclass
+from .weapon import Weapon, Blacksmith
+from .subclasses import (
+    Champion,
+    EldritchKnight,
+    BattleMaster,
+    BeastMaster,
+    Hunter,
+    Subclasses,
+)
+from .classes import Class
+from .utilities import (
+    proficiency_bonus_by_level,
+    chance_to_hit,
+    chance_of_an_instance,
+    chance_to_critical,
+)
+import json
+
+SMITH = Blacksmith()
+
+
+@dataclass
+class Character:
+    weapon: Weapon
+    bonus_weapon: Weapon
+    subclass: Class
+    attack_stat: int
+    advantage: bool
+    level: int
+    proficiency_bonus: int
+    enemy_armour_class: int
+    dual_wielder: bool
+    sharpshooter: bool
+    great_weapon_master: bool
+
+    def __init__(self, data):
+        bonuses = json.loads(data["bonuses"])
+        feats = json.loads(data["feats"])
+        self.level = int(data["characterLevel"])
+        self.proficiency_bonus = proficiency_bonus_by_level(self.level)
+        self.enemy_armour_class = int(data["averageAC"]) if data["averageAC"] else 0
+        if data["subclass"] == Subclasses.CHAMPION:
```

Comment:
Similar thing in the `Blacksmith` class, I don't really like all these if elif statements. I'm not completely sure how to change it, I think it would involve moving responsibilities a bit within classes. I'll have a think, it's ultimately not a problem, it just seems like it should be unecessary.